js/modules/Search/Table.js

```diff
@@ -151,6 +153,35 @@ window.GLPI.Search.Table = class Table extends GenericView {
         this.refreshResults();
     }
 
+    onSortContainerChange(sort_container) {
+        const sorts = [];
+        const orders = [];
+
+        $(sort_container).find('select[name^="order"]').each(function() {
+            orders.push($(this).val());
+        });
+        $(sort_container).find('select[name^="sort"]').each(function() {
+            sorts.push($(this).val());
+        });
+
+        // Remove data-sort-num from all headers and change all data-sort-order to nosort to reset the sort state
+        this.getElement().find('thead th').each((i, c) => {
+            const col = $(c);
+            col.attr('data-sort-num', null);
+            col.attr('data-sort-order', 'nosort');
+        });
```

Comment:
FYI jQuery's `attr` method work on the whole collection, there is no need to iterate on each items.

Something like that would work for example:

```suggestion
        this.getElement().find('thead th').attr({
            'data-sort-num': null,
            'data-sort-order': 'nosort',
        });
```